velox/functions/sparksql/String.h

```diff
@@ -1431,4 +1433,143 @@ struct LevenshteinDistanceFunction {
   }
 };
 
+/// split(string, delimiter[, limit]) -> array(varchar)
+///
+/// Splits string on delimiter and returns an array of size at most limit.
+/// delimiter is a string representing regular expression.
+/// limit is an integer which controls the number of times the regex is applied.
+/// By default, limit is -1.
+template <typename T>
+struct SplitFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
```

Comment:
An initialize method could be added to handle constant delimiter and limit.

Reply:
any example code can reference ? @rui-mo , thanks

Reply:
I understand. I agree that initialize cannot bring much benefits as the processing is row by row.

Reply:
ping @rui-mo for above help ask.

Reply:
Some documentation on constant inputs: https://github.com/facebookincubator/velox/blob/main/velox/docs/develop/scalar-functions.rst#access-to-session-properties-and-constant-inputs.

Reply:
i saw this part doc but still not sure how to leverage do optimize @rui-mo

Reply:
I double-checked and found initialize cannot bring much benefits for this function, seeing https://github.com/facebookincubator/velox/pull/10248#discussion_r1689372828.